### PR TITLE
WIP - 6.0.0

### LINKED ISSUE #236: v6
This will be the biggest update since v1. Gonna go all out here.

- [ ] Figure out ideal testing setup #277
- [ ] Updated CLI #263
- [x] Look into a build tool that doesn't rely on Acorn for ES7 features #230 

and more! Those are the big ones tho'

### PR DESCRIPTION
**Pull request by jmeas**
Will resolve #236 

Todo:

- [x] Rebase off https://github.com/babel/generator-babel-boilerplate/pull/289
- [x] Final testing
- [x] Release

### CODE REVIEW

File: app/templates/gulpfile.babel.js [hunk 1]
```diff
@@ -154,26 +115,57 @@ function coverage(done) {
...
+  // should individually load up pieces of our application.
+  // We also include the browser setup file.
+  const testFiles = glob.sync('./test/unit/**/*.js');
+  const allFiles = ['./test/setup/browser.js'].concat(testFiles);
+
+  // Lets us differentiate between the first build and subsequent builds
+  var firstBuild = true;
+
+  // This empty stream might seem like a hack, but we need to specify all of our files through
+  // the `entry` option of webpack. Otherwise, it ignores whatever file(s) are placed in here.
```
Comment (thejameskyle) [inline app/templates/gulpfile.babel.js]: Why don't you just start with the `webpackStream` then?

Comment (jmeas) [inline app/templates/gulpfile.babel.js]: 'cuz I'm a noob. I'll try that

**Comment by jmeas** [comment]
I just realized that simply using `gulpfile.babel.js` removes the need to ever run `_registerBabel`, as Babel gets registered automatically with Babel'd gulpfiles. So that's something I can update to clean up the Gulpfile.

**Comment by thejameskyle** [comment]
Hmm... The way gulp should probably be implementing that is to limit the register to only the gulpfile.

**Comment by jmeas** [comment]
> Hmm... The way gulp should probably be implementing that is to limit the register to only the gulpfile.

Yeah? I wonder if it's a bug, then. It's working [over here](https://github.com/jmeas/testing-with-babel/tree/master/examples/mocha-gulp).

**Comment by jmeas** [comment]
Alright, finally gonna cut this release then start work on the next version, which will use Babel v6!

**Pull request merged**